pkg/lang/javascript/aws_runtime/fs.js.tmpl

```diff
@@ -88,15 +88,24 @@ async function s3_writeFile(...args) {
 async function s3_readFile(...args) {
     const bucketParams = {
         Bucket: bucketName,
-        Key: `${args[0]}`,
+        Key: stripLeadingSlashes(`${args[0]}`),
     };
     try {
         // Get the object from the Amazon S3 bucket. It is returned as a ReadableStream.
         const data = await s3Client.send(new client_s3_1.GetObjectCommand(bucketParams));
         if (data.Body) {
-            return await streamToString(data.Body);
+            if (args[1]?.encoding) {
+                return await streamToString(data.Body);
+            }
+            return new Promise((resolve, reject) => {
+                const stream = data.Body;
+                const chunks = [];
+                stream.on('data', chunk => chunks.push(chunk))
+                stream.once('end', () => resolve(Buffer.concat(chunks)))
+                stream.once('error', reject)
+            });
```

Comment:
nit: make this reusable? Looks similar to steamToString as well but maybe nothing to pull out between the 2

Reply:
Simple enough. I hadn't even looked at the content of `streamToString`.